tools/tpm2_nvwrite.c

```diff
@@ -61,13 +61,16 @@ struct tpm_nvwrite_ctx {
     UINT16 offset;
     char *raw_pcrs_file;
     tpm2_session *policy_session;
+    tpm2_session *hmac_session;
+    bool hmac_auth;
     TPML_PCR_SELECTION pcr_selection;
     struct {
         UINT8 L : 1;
     } flags;
 };
 
 static tpm_nvwrite_ctx ctx = {
+    .hmac_auth = false,
```

Comment:
I wonder if we want the default to be true and instead have an option to use plaintext passwords? My understanding is that not using HMAC sessions is insecure so probably makes sense to do it that way?

Reply:
@martinezjavier @williamcroberts , then we may need to account for if hmac sessions are not available. any thoughts?

Reply:
@idesai oh, isn't HMAC supported on all TPM2 devices or am I not following?

Reply:
When are hmac sessions not available?

I think the default auth should always be a NULL password. Then users use the auth mini-language to specify how they want to auth.

ie:
```
--auth-xxx password
Create an hmac ticket
--auth-xxx hmac:

Reply:
@martinezjavier @williamcroberts by sessions not available, I meant max allowed sessions already resident on the TPM. Starting a new session for hmac would result in TSS2_RC "out of memory"

Reply:
Oh yeah, well that's always a potential issue when using the TPM directly. It's a user problem if their using the TPM without an RM and get this error code.

Reply:
@idesai ah, got it. Hmm, I don't think the tools should handle that. I mean, you could also attempt to create/load more objects than TPM_PT_HR_TRANSIENT_MIN and the TPM (and the tools) would just return a TPM_RC_OBJECT_MEMORY in that case.

So I think that we should just make HMAC session auth the default and if the user is getting this, then it should either access the TPM through a resource manager or use plain text passwords. Just like you need to either manage your objects or use a resource manager.

What do you think?

Reply:
@williamcroberts I see that you agree (you answered while I was writing mine :P)

Reply:
@martinezjavier @williamcroberts, cool lets treat it as such. 
I like the tpm2_auth_util_from_optarg way of doing this which should address it.
-P hmac:<Password>

Reply:
The tpm2_auth_util_from_optarg was added recently, So I have to rebase this PR.

Reply:
@martinezjavier @williamcroberts I added a new routine tpm2_auth_util_from_optarg_new to parse hmac: string. If this is good then we could change it in other tools.